### DISCUSSION TITLE
Mission based XP gains

### DISCUSSION THREAD

**Discussion by ghost** [FakeFishGames/Barotrauma#7288] [Ideas and suggestions]
Currently I don't think what missions you choose actually affect the XP you get at the end of the level, I'd like to propose making XP part of mission rewards, since it makes a lot of sense you'd get more XP for clearing multiple nests than transporting a single box of near worthless value.

**Comment by StruggleDuck** [comment]
as it is right now, the amount of xp you get is actually linked to how much money you made that round (I'm fairly confident this is the case)

**Comment by mLuby** [comment]
Respectfully, I oppose mission-based XP.

The mechanic shouldn't gain more experience from a cave nest mission and less experience from a transport mission; instead the mechanic should gain more experience from performing actions during rounds and less experience from doing the same old thing over and over again. (looking at you, fabricator jockeys).

So if the cave nest mission is relatively uneventful but during the transport mission a thalamus grabbed the sub and it was only saved by the mechanic repairing the breaches while fighting off [Terminals Cells](https://barotraumagame.com/wiki/Terminal_Cells), the transport mechanic deserves more XP (and a commendation)! 🎖️

**Comment by MaaiG** [comment]
In that case, account should be taken not only of participation in the mission, but also of the specific contributions of each participant. If the mechanic has repaired 10 pumps and 6 of them in the flooded compartments, he must receive his medal.
In view of the amount of work he’s been put in because of the Captain’s stupidity, etc. By the way, the Captain’s glasses will have to be removed ))

By the way, consider the number and complexity of unscheduled repairs!
And not when he broke it himself. For damage - bonuses are removed. ...And a fine is imposed! $)
